src/fsharp/TypeChecker.fs

```diff
@@ -5325,6 +5325,15 @@ and RecordNameAndTypeResolutions_IdeallyWithoutHavingOtherEffects_Delayed cenv e
         | _ -> ()
     dummyCheckedDelayed delayed 
 
+// Calls UnifyTypes, records name resolutions on failure, so that type information
+// can be collected for IntelliSense.
+and UnifyTypes_RecordNameResolutionsOnError cenv env m expectedTy actualTy =
```

Comment:
Done. I hadn't done any cleanup after my excitement in getting something working!